### ISSUE TITLE
JAR available on Maven Central

### ISSUE THREAD

**Issue by bmuschko** [tinyeeliu/androidquery#12]
Hi,

I'd like to reference your library from a Maven-compatible repository. Do you have any plans to upload each version to Maven Central or another OSS repo?

Thanks,

Ben

**Comment by tinyeeliu** [comment]
Hi.

That's definitely something we want to do with AQuery.

Do you have any specific detail (like instructions or guide) on uploading libs to the platforms you mentioned? That will accelerate the process.

Thanks.

**Comment by bmuschko** [comment]
Personally, I'd prefer building the project with [Gradle](http://gradle.org/). I am fairly new to Android development but I tried out the [Android plugin](https://github.com/jvoegele/gradle-android-plugin) recently which would allow you to build the project. For uploading AQuery to Maven you can simply use Gradle as well. See this [tutorial](http://jedicoder.blogspot.com/2011/11/automated-gradle-project-deployment-to.html) for more information. 

Of course all of this can be done with Maven as well but I think Gradle is far more expressive and superior in its capabilities.

**Comment by adennie** [comment]
Here's a detailed tutorial on uploading to the Maven Central repo: https://docs.sonatype.org/display/Repository/Sonatype+OSS+Maven+Repository+Usage+Guide

**Comment by ClintEsteMadera** [comment]
I'd also strongly appreciate if the library could be uploaded to a maven repository. My android projects are all maven-enabled. Thanks very much in advance.

**Comment by adrian-chang** [comment]
Upvote! Please add if possible.

**Comment by ghost** [comment]
You can manually add the jar ONCE to the maven repo on your local machine using the following

mvn install:install-file -Dfile=android-query.jar -DgroupId=android-query -DartifactId=ta-lib -Dversion=1.0.0 -Dpackaging=jar

**Comment by ClintEsteMadera** [comment]
yeah, that's what I did but it is a work around the problem. I access my project from different computers and i would need to do that ONCE per different COMPUTER. It would be much better to upload the artifact to central...thanks for the advice!

**Comment by davidschreiber** [comment]
Would also like to see a Maven repository for Androidquery! Upvote this!

**Comment by tinyeeliu** [comment]
Need some help on setting this up.

Please check
https://groups.google.com/forum/?fromgroups=#!topic/android-query/MytwIFv-cjI

**Comment by brk3** [comment]
https://github.com/androidquery/androidquery/pull/41

**Comment by vbauer** [comment]
Need it too! :thumbsup:

**Comment by bachi76** [comment]
With Gradle becoming the standard solution for Android development, it would be superb if http://mvnrepository.com/artifact/com.googlecode.android-query/android-query could be updated to the latest version.

**Comment by jendib** [comment]
@bachi76 +1 it would be great.

**Comment by Gryzor** [comment]
:+1:

**Comment by charafau** [comment]
agree! some integration server with auto release should be nice ;)

**Comment by Sereri** [comment]
I'm not sure about auto releases but at least updating maven from 25.9 to not-broken release 25.10 would be nice
